plugins/nvme/fu-nvme-device.c

```diff
@@ -245,6 +247,18 @@ fu_nvme_device_set_serial(FuNvmeDevice *self, const gchar *serial, GError **erro
 						 "SNSUFFIX",
 						 NULL))
 			return FALSE;
+		model_name = fu_device_get_instance_str(FU_DEVICE(self), "NAME");
+		if (model_name != NULL) {
```

Comment:
you dont need to do this; just launch right into the fu_device_build_instance_id -- any missing keys return an error - so we can ignore that with setting the error to NULL and just not checking the return value. e.g. https://github.com/fwupd/fwupd/blob/main/libfwupdplugin/fu-udev-device.c#L421